assets/js/hocs/with-single-selected.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { Component } from '@wordpress/element';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import PropTypes from 'prop-types';
+import { isNil } from 'lodash';
+
+const withSingleSelected = createHigherOrderComponent(
+	( OriginalComponent ) => {
+		class WrappedComponent extends Component {
+			render() {
+				const { selected } = this.props;
+
+				return (
+					<OriginalComponent
+						{ ...this.props }
+						selected={ [ selected ].filter(
+							( val ) => ! isNil( val )
+						) }
```

Comment:
I'm a little puzzled by this.  From the name of the HOC, I'd expect that `withSingleSelected` ensures there's only one selected value in an array, but from the logic it appears that all it does is filter out any non nil values from the provided array?  Either the logic isn't quite correct or the naming of this HOC can be improved?

Reply:
`withSingleSelected` converts a `selected` value into an array, so it's compatible with other HOCs that rely on `selected` being an array. I refactored the code above to make it more obvious what it does: fd1ed3263b997d2cfbc83546b0ceb90a184fb2ea.

Reply:
Ahh.  Makes sense, however I think we can try to make the HOC named a bit better. At the risk of verboseness, how about:

- withSingleSelectTransform
- withTransformSingleSelectToMultipleSelect
- withEnsureSelectValueIsArray

Open to other options as well. I think we should aim for a name that's a bit more descriptive of the purpose of the HOC though.

Reply:
Sounds good, I renamed it to `withTransformSingleSelectToMultipleSelect`, a bit long but probably the one that better describes what the HOC does: 9ead24c7560805916dd46bcc155bf15867bdd2e8.